test/infra/control/run-multi-group.bash

```
@@ -0,0 +1,423 @@
+
+# Function to run a single group
+run_single_group() {
+    local group="${1}"
+    local infra_id="${group}-${RUN_ID}"
+    local log_file="${RESULTS_DIR}/${group}.log"
+    local start_time end_time duration
+
+    # Add random delay to stagger infrastructure startup
+    # This prevents resource contention when running multiple groups in parallel
+    # Use group name hash + random to ensure both unique and unpredictable delays
+    local group_hash=$(echo -n "${group}" | cksum | cut -d' ' -f1)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)